### DISCUSSION TITLE
How to define channel model in ray tracing scenario

### DISCUSSION THREAD

**Discussion by YuruZhang22** [NVlabs/sionna#460] [Q&A]
Thanks for your amazing work!

I am trying to adjust some parameters to change the channel model so that I can obtain different CIRs from the same position, which I use to calculate the RSRP.

I have found that I can change the material of the objects in the scene to get different reflection coefficients. Additionally, I can change the max_depth parameter to limit the ray's interactions with scene objects.

I want to ask if there are other methods that allow me to define the channel model in the ray tracing scenario. For example, is there a way to define the channel attenuation model, or does the tool use a default channel model?

**Comment by SebastianCa** [comment]
Hi @YuruZhang22,

I’m not entirely sure if I understand your experiment correctly. Typically, you either change the user position or simulate stochastic effects, such as thermal noise, to draw different channel realizations at the same position.

Keep in mind that `max_depth` is a simulation parameter and only defines the accuracy of your simulation (i.e., the ray tracer discards every path with more than `max_depth` reflections). In the real world, these effects still exist, so the results may not be entirely meaningful for your experiment.

There are several parameters that could be adjusted, such as material properties, carrier frequency, antenna patterns, and the position of scene objects. For further details, please see the [ray tracing tutorial](https://nvlabs.github.io/sionna/examples/Sionna_Ray_Tracing_Introduction.html). You could also consider [Diffraction](https://nvlabs.github.io/sionna/examples/Sionna_Ray_Tracing_Diffraction.html) and [Scattering](https://nvlabs.github.io/sionna/examples/Sionna_Ray_Tracing_Scattering.html).

**Comment by YuruZhang22** [comment]
Hi @SebastianCa,

Thank you for your feedback.

We implemented a real-world scenario in Sionna and aim to calculate the RSRP of certain locations using ray tracing. We used the following calculation method: rsrp = -20*np.log10(np.abs(a)), where a and tau are derived from the path's CIR. However, we've observed discrepancies between the RSRP measured in the real world and the RSRP calculated using Sionna RT for the same location.

To address these discrepancies, we are exploring ways to adjust the parameters within Sionna RT to align the calculated RSRP with real-world measurements.

We found the passband channel coefficients a_i for each path as defined in equation (26) of the Sionna documentation (https://nvlabs.github.io/sionna/em_primer.html#frequency-impulse-response). However, some parameters such as C_R, T_i, and C_T are not entirely clear to us. Can these parameters be adjusted, or are they fixed within the simulation framework?

Additionally, are there other parameters that we should consider adjusting to improve the accuracy of our simulation results? We cannot change the carrier frequency, antenna patterns, or the position of scene objects as these parameters must remain consistent between the real-world scenario and the Sionna simulation. We would appreciate any guidance or references to further documentation, such as the ray tracing tutorial or considerations for diffraction and scattering effects.

Thank you for your assistance.

Best regards,
Yuru Zhang

**Reply by SebastianCa** [reply]
There are a couple of possible reasons for the observed mismatch:
- Most likely the scene modelling is inaccurate (i.e., the mesh does not exactly fit reality)
- The exact material properties of objects are unknown 
- Not all effects are considered in the ray tracer (e.g., refraction)
- The measured dataset might also have some noise

A possible solution is to use machine learning to *calibrate* the material properties such that the ray tracer output equals to measured results. For a more detailed discussion about the above points and some code examples, please see the [differentiable ray tracing repository](https://github.com/NVlabs/diff-rt) and the [differentiable ray tracing paper](https://arxiv.org/pdf/2311.18558).

I hope this helps.

**Reply by YuruZhang22** [reply]
Thank you for your detailed response. Your insights into the possible reasons for the observed mismatch are very helpful, and we will certainly explore the use of machine learning to calibrate material properties as suggested.

However, we still have some specific questions regarding the calculation of the passband channel coefficients a_i as defined in equation (26) of the documentation (https://nvlabs.github.io/sionna/em_primer.html#frequency-impulse-response). Specifically, we are unclear about the meaning and adjustability of the parameters C_R, T_i, and C_T. Can these parameters be adjusted, or are they fixed within the simulation framework? Due to my limited knowledge of electromagnetic fields, I am particularly interested in understanding the significance and roles of these three parameters in the calculation.

Your guidance on this matter would be greatly appreciated.Thank you again for your assistance.

Best regards,  
Yuru Zhang

**Comment by SebastianCa** [comment]
`C_T` and `C_R` represent the transmit and receive antenna patterns, respectively. These can be customized; see [Antennas](https://nvlabs.github.io/sionna/api/rt.html#antennas) for details. Ensure these parameters match your real-world hardware.

`T_i` is the transfer matrix, capturing all effects during ray propagation, and indirectly depends on the scene geometry and material definitions.

I hope this clarifies the meaning of the parameters.

**Reply by YuruZhang22** [reply]
Thank you for your clarification regarding the parameters. That helps a lot.

As you mentioned that, the effects of ray propagation as well as the materials are included in the transfer matrix [T]. But, how the distance-related attenuation is considered during the ray propagation. In the source code, I saw that the value of [a_i] is calculated by the .cir(), which only relates to [C_t, T, C_r], but no distances between Tx and Rx antenna. Take a simple example, if Tx and Rx has LOS in free space, then the [a_i] has only one value in a LOS delay. If Tx-Rx distance is 1 meter, then the [a_i] should be much higher than that of 10 meters Tx-Rx distance, right? In this case, how does this distance-related attenuation reflected in Sinnoa?

Thank you again for your assistance.

Best regards,  
Yuru Zhang

**Comment by SebastianCa** [comment]
That’s correct, the distance-dependent attenuation is part of the transfer matrix. You can find details about free space path loss in [Eq. 15 in the EM Primer](https://nvlabs.github.io/sionna/em_primer.html#equation-e-t) and the [example on the Friis equation just below Eq. 21](https://nvlabs.github.io/sionna/em_primer.html#equation-v-r).

**Reply by YuruZhang22** [reply]
Thank you for your response.

I would like to know more about where in the code this transfer matrix **T** is defined and how it can be adjusted. Specifically, which parameter within **T** represents the distance or is related to the distance between the Tx and Rx antennas?

Thank you again for your assistance.

Best regards,  
Yuru Zhang

**Comment by SebastianCa** [comment]
Please note that you should not modify the outcome of the ray tracer to align the RSRP (and similar metrics). Instead, you should *calibrate* the simulation parameters so that the modeled behavior matches the measured results. This implies changing one or more of the following:

	•	The underlying scene geometry (e.g., the position of objects, which affects the attenuation due to different path lengths)
	•	The material properties
	•	The antenna patterns

For further details about the ray tracing output, please refer to the code related to [compute paths](https://nvlabs.github.io/sionna/api/rt.html#compute-paths).
